code/tailer/datastripe.q

```diff
@@ -124,28 +143,27 @@ savetablesoverperiod:{[dir;tablename;nextp;lasttime]
 
     /- enumerate and then split by keycol
     symdir:` sv dir,.proc.procname;
-    enumkeycol: .Q.en[symdir;?[tablename;enlist (<;`time;nextp);0b;()]];
+    enumkeycol: .Q.en[symdir;eval tablename];
     splitkeycol: {[enumkeycol;keycol;s] ?[enumkeycol;enlist (=;keycol;enlist s);0b;()]}[enumkeycol;keycol;] each partitionlist;
 
     /-upsert table to partition
-    upserttopartition[dir;tablename;keycol;;nextp] each splitkeycol where 0<count each splitkeycol; 
-
-    /- delete data from last period
-    .[.ds.deletetablebefore;(tablename;`time;lasttime)];
+    upserttopartition[dir;tablename;keycol;] each splitkeycol where 0<count each splitkeycol; 
     
     /- run a garbage collection (if enabled)
     .gc.run[];
     };
 
-savealltablesoverperiod:{[dir;nextp;lasttime]
+savealltablesoverperiod:{[dir;lasttime]
     /- function takes the tailer hdb directory handle and a timestamp
     /- saves each table up to given period to their respective partitions
-    savetablesoverperiod[dir;;nextp;lasttime]each .wdb.tablelist[];
+    savetables[dir;]each .wdb.tablelist[];
```

Comment:
I could definitely be missing something here but...

If we're not passing `lasttime` to the save function, how do we guarantee that the data that's deleted in line 162 is the same as what's been saved? it seems to me like we'll save everything in the table, but then only delete the data before `lasttime`, so the next time we save we could be duplicating some data that was saved last time we saved? (e.g. if there was already some data where `time>lasttime` when we did the first save, it wouldn't get deleted and then would be saved again in second save)

Reply:
The delete on line 162 is passed lasttime, but that function is savealltablesoverperiod and and is only used by eop. Savetables is the function above and only saves the tables. EOD func and savealltablesoverperiod use the savetables function to save tables and then tables are deleted in different ways afterwards by eod and savealltablesoverperiod.

Reply:
OK sure - but my point is, if `savetables` doesn't know about any limitations on what to save, surely it just saves everything in the table? So should line 162 just delete everything in the table? What I'm struggling to understand is if we save everything, but then only delete data before a certain time within this same function (`savealltablesoverperiod`)? And if so, why?

Reply:
There is another PR which has it changed so that after the savedown the contents of the tablelist tables are deleted, so that should resolve this issue.

Reply:
So we have two open PRs atm that are modifying this function? Should we remove the changes to this function in one or other PR? I assume we're going to get a conflict if we try to merge both

Reply:
Yeah I am talking to the guys about it now. There was a bit of scope creep on one ticket that ended-up affecting the other, so will make sure the second PR is adapted to work with David's. The other PR needs reworking in various ways anyway before it is ready for review again anyway.

Reply:
I see this conversation is resolved - but what's the resolution?

Reply:
For eod the data is saved and then all in mem tables are deleted by line 48 it does not use deletealltablesbefore. The deletealltablesbefore that uses lasttime is used by eop and is already merged into torq5. So would this not need to be a separate ticket to investigate?

Reply:
I have updated savealltables so that it now just removes all data that has been saved both at eop and eod

Reply:
Looks good now